src/core/ddsi/src/q_entity.c

```diff
@@ -4475,9 +4475,6 @@ dds_return_t delete_reader (struct ddsi_domaingv *gv, const struct ddsi_guid *gu
   GVLOGDISC ("delete_reader_guid(guid "PGUIDFMT") ...\n", PGUID (*guid));
   builtintopic_write (rd->e.gv->builtin_topic_interface, &rd->e, ddsrt_time_wallclock(), false);
   entidx_remove_reader_guid (gv->entity_index, rd);
-#ifdef DDSI_INCLUDE_SECURITY
```

Comment:
I don't understand why this should disappear

Reply:
The call to q_omg_security_deregister_reader at this location could cause a crash because the sec_attr field could still be accessed at that moment. The function gc_delete_reader also calls q_omg_security_deregister_reader where it is safe.

Reply:
Ah, ok, that makes sense. Doesn't the same argument apply to ``q_omg_security_deregister_writer``?

Reply:
Yes you are right I missed that one will correct it.